include/sound/sof.h

```diff
@@ -90,6 +90,8 @@ struct snd_sof_pdata {
 	const struct snd_soc_acpi_mach *machine;
 
 	void *hw_pdata;
+
+	enum sof_ipc_types ipc_type;
```

Comment:
Looks great @plbossart . The only opens I have are that the dsp ops partitioned in the first patch need to also be made of sice SOF_IPC_COUNT, right ? And the way you've defined the cavs IPC, it can't use the SOF firmware. Is that ok? What happens when SOF firmware starts supporting IPC4

Reply:
> The only opens I have are that the dsp ops partitioned in the first patch need to also be made of size SOF_IPC_COUNT, right ? 

> And the way you've defined the cavs IPC, it can't use the SOF firmware. Is that ok? What happens when SOF firmware starts supporting IPC4

 The idea for this PR is that for each platform there is a default IPC version, which can be overridden manually by developers. If there are multiple implementations of the IPCv4, I think again we'll have to select a default path and manually override if we want to switch between implementations. For example, instead of 'intel/avs' the path would be 'intel/sof-ipcv4'. The file names and topologies would have to be the same though.

Reply:
> Not following this one @ranj063, can you explain what you are referring to?
> 
> >

https://github.com/thesofproject/linux/blob/9ea6ec88b75bae1bb07a24fd0d254da4a76f97c7/sound/soc/sof/intel/cnl.c#L246

I was wondering if the array above needs to be of size SOF_IPC_COUNT too

Reply:
yes, nice catch indeed. It only works today because I only added the default index 0, but that wouldn't work if I want to define only INTEL_CAVS, as planned for SKL/KBL.

Thanks for spotting this @ranj063

Reply:
fixed